envify_generator/lib/src/helpers/string.dart

```diff
@@ -2,26 +2,26 @@
 
 /// Type Definition: function with a [String] as a parameter
 /// returning a [bool]
-typedef boolFn = bool Function(String);
+typedef boolFn = bool Function(String?);
 
 /// Method to decide whether a [String] input
 /// returns `true` or `false` through the predicate function
 /// or not.
-bool validate(String input, boolFn predicate) {
+bool validate(String? input, boolFn predicate) {
   return predicate(input);
 }
 
 /// Method that will returns
 /// [a] if [a] is not null or an empty string. Otherwise,
 /// [b] will be returned.
-String ifFalsy(String a, String b) {
+String? ifFalsy(String? a, String b) {
   return validate(a, (x) => (x ?? '').isNotEmpty) ? a : b;
```

Comment:
Dart won't infer type but here `a` is not nullable, we might need to make it non-nullable manually as `a!`.

Reply:
Hm, but as I see code for the `annotation.read('name').literalValue as String?` it can be nullable by it's nature. 

`Object? get literalValue => null;` 

So, I think we should account for that to achieve null safety?

Reply:
The typecasting there is I guess meant for typecasting from `Object?` to `String?`. I don't think it has anything to do with its nullability. Plus `validate(a, (x) => (x ?? '').isNotEmpty)` ensures that `a` can't be null.